### ISSUE TITLE
Query Batching

### ISSUE THREAD

**Issue by miguelocarvajal** [graphql-java/graphql-java-servlet#43]
Using GraphQL-Servlet 4.4

I am getting the following exception when using batching with the Apollo client:

com.fasterxml.jackson.databind.exc.MismatchedInputException: Can not deserialize instance of graphql.servlet.GraphQLServlet$GraphQLRequest out of START_ARRAY token
 at [Source: (org.apache.catalina.connector.CoyoteInputStream); line: 1, column: 1]
	at com.fasterxml.jackson.databind.exc.MismatchedInputException.from(MismatchedInputException.java:62) ~[jackson-databind-2.9.0.pr4.jar:2.9.0.pr4]
	at com.fasterxml.jackson.databind.DeserializationContext.reportInputMismatch(DeserializationContext.java:1307) ~[jackson-databind-2.9.0.pr4.jar:2.9.0.pr4]
	at com.fasterxml.jackson.databind.DeserializationContext.handleUnexpectedToken(DeserializationContext.java:1116) ~[jackson-databind-2.9.0.pr4.jar:2.9.0.pr4]
	at com.fasterxml.jackson.databind.DeserializationContext.handleUnexpectedToken(DeserializationContext.java:1070) ~[jackson-databind-2.9.0.pr4.jar:2.9.0.pr4]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromArray(BeanDeserializerBase.java:1440) ~[jackson-databind-2.9.0.pr4.jar:2.9.0.pr4]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeOther(BeanDeserializer.java:185) ~[jackson-databind-2.9.0.pr4.jar:2.9.0.pr4]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:161) ~[jackson-databind-2.9.0.pr4.jar:2.9.0.pr4]
	at com.fasterxml.jackson.databind.ObjectReader._bindAndClose(ObjectReader.java:1613) ~[jackson-databind-2.9.0.pr4.jar:2.9.0.pr4]
	at com.fasterxml.jackson.databind.ObjectReader.readValue(ObjectReader.java:1183) ~[jackson-databind-2.9.0.pr4.jar:2.9.0.pr4]
	at graphql.servlet.GraphQLServlet.lambda$new$1(GraphQLServlet.java:163) ~[graphql-java-servlet-4.4.0.jar:na]
	at graphql.servlet.GraphQLServlet.doRequest(GraphQLServlet.java:229) ~[graphql-java-servlet-4.4.0.jar:na]
	at graphql.servlet.GraphQLServlet.doPost(GraphQLServlet.java:247) ~[graphql-java-servlet-4.4.0.jar:na]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:661) ~[tomcat-embed-core-8.5.16.jar:8.5.16]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:742) ~[tomcat-embed-core-8.5.16.jar:8.5.16]

..snip..

Is query batching supported?

**Comment by apottere** [comment]
Query batching isn't supported yet, as I didn't know it existed :)  Would you mind providing a sample request that should be handled?

**Comment by miguelocarvajal** [comment]
Sure thing! We're using the Apollo client which batches the requests like this (formatted this as best as I could):

```
[
  {
    "query":"
      query ($level: Int) {
        treeTest(level: $level) {
          pk
          name
          gender
          primaryEmail {
            address
            __typename
          }
          primaryAddress {
            address1
            __typename
          }
          __typename
        }
      }
   ",
   "variables":{
     "level":-1
    },
   "operationName":null
  }
]
```

Here is their documentation regarding batching: http://dev.apollodata.com/core/network.html#query-batching

If there's anything else you need just let me know.

Thanks!

**Comment by apottere** [comment]
That looks pretty easy, I'll see what I can do.  Thanks for the example!

**Comment by ronald-d-rogers** [comment]
Just wondering if any progress has been made on this.

**Comment by ronald-d-rogers** [comment]
@apottere Do you mind if a I take a stab at this? This is kind of an important feature for us.

**Comment by ronald-d-rogers** [comment]
I have this completed and tested, PR incoming soon.

**Comment by ronald-d-rogers** [comment]
Completed (See: #48)

**Comment by apottere** [comment]
Fixed in #48 (thanks @ronald-d-rogers!)

**Comment by apottere** [comment]
This is syncing to central in `4.5.0` now.
